src/main/java/echoserver/ClientConnection.java

```diff
@@ -0,0 +1,40 @@
+package echoserver;
+
+import java.io.*;
+import java.net.Socket;
+
+public class ClientConnection {
```

Comment:
I am curious about this interface. It seems very specific, and not much of it seems to be in the domain of a *connection*

Reply:
It looks like there are also varying levels of abstraction. 

`promptInput`, `readLastSavedInput`

vs. 

`getOutputStream`

vs

`isClosed`